lms/views/dashboard.py

```diff
@@ -127,6 +178,20 @@ def get_request_assignment(self):
 
         return assignment
 
+    def get_request_course(self):
+        course = self.course_service.search(self.request.matchdict["id_"])
+        if not course:
+            raise HTTPNotFound()
+
+        if self.request.has_permission(Permissions.STAFF):
+            # STAFF members in our admin pages can access all assignments
+            return course[0]
+
+        if not self.course_service.is_member(course, self.request.user):
```

Comment:
course_service.is_member is not implemented.

This only works while accessing from the admin pages.